notebooks/proj_sys_eng/sitcom_integration/l3_system_integ/LVV-T2216-plots.ipynb

```diff
@@ -0,0 +1,1166 @@
+    "print(t_end, df.iloc[0][\"message\"])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c21f65ea-da47-4612-9846-aa9da06e7479",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "print(f\"Starting data analysis between\\n  {t_start}\\n    and\\n  {t_end}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2cd8a712-02ce-4427-aad8-0a8c428e8100",
+   "metadata": {},
+   "source": [
+    "### Analyse M1M3 Data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "eaaa7d8c-5c07-45f3-be20-5d0eb4482a81",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fad = await client.select_time_series(\n",
+    "    \"lsst.sal.MTM1M3.forceActuatorData\", \n",
+    "    \"*\", \n",
+    "    t_start,\n",
+    "    t_end)\n",
+    "\n",
+    "el = await client.select_time_series(\n",
+    "    \"lsst.sal.MTMount.elevation\",\n",
+    "    \"*\", \n",
+    "    t_start,\n",
+    "    t_end)\n",
+    "\n",
+    "fig, axs = plt.subplot_mosaic(\n",
+    "    mosaic=\"A\",\n",
+    "    num=\"Slew Without Correction\", \n",
+    "    constrained_layout=True,\n",
+    "    dpi=120,\n",
+    "    figsize=(12, 4),\n",
+    ")\n",
+    "\n",
+    "fig.suptitle(\"M1M3 Timeline - Slew LUT+inclinometer\")\n",
+    "_ = vandv.m1m3.timeline_zforces(axs[\"A\"], fad, \"zForce\", elevation=el)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bcc01609-dc8c-48db-bcd9-f4ec15f81949",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fel = await client.select_time_series(\n",
+    "    \"lsst.sal.MTM1M3.logevent_appliedElevationForces\", \n",
+    "    \"*\", \n",
+    "    t_start,\n",
+    "    t_end)\n",
+    "\n",
+    "fba = await client.select_time_series(\n",
+    "    \"lsst.sal.MTM1M3.logevent_appliedBalanceForces\", \n",
+    "    \"*\",\n",
+    "    t_start,\n",
+    "    t_end)\n",
+    "\n",
+    "fst = await client.select_time_series(\n",
+    "    \"lsst.sal.MTM1M3.logevent_appliedStaticForces\",\n",
+    "    \"*\",\n",
+    "    t_start,\n",
+    "    t_end)\n",
+    "\n",
+    "fao = await client.select_time_series(\n",
+    "    \"lsst.sal.MTM1M3.logevent_appliedActiveOpticForces\",\n",
+    "    \"*\", \n",
+    "    t_start,\n",
+    "    t_end)\n",
+    "\n",
+    "series = [df.iloc[-1] for df in [fel, fba, fst, fad]]\n",
+    "labels = [\n",
+    "    \"appliedElevationForces\", \n",
+    "    \"appliedBalanceForces\", \n",
+    "    \"appliedStaticForces\",\n",
+    "    \"forceActuatorData\",\n",
+    "]\n",
+    "\n",
+    "fig, axs = plt.subplot_mosaic(\n",
+    "    mosaic=\"AB\\nAB\\nCC\\nCC\",\n",
+    "    num=\"Slew Without Correction - Snapshot Forces\", \n",
+    "    constrained_layout=True,\n",
+    "    dpi=120,\n",
+    "    figsize=(9, 5),\n",
+    ")\n",
+    "\n",
+    "fig.suptitle(\"M1M3 Snapshot - LUT+inclinometer - XYZ end slew\")\n",
```

Comment:
This snapshots here are calculated at the end of the slew t_end. . It would be great if the time could be selected, for future comparisons.

Reply:
You can select the time by using `df.loc[timestamp]`. I left it separated as a design choice. This way, the data selection and the data display are uncoupled, and it is easier to re-use the code.